server/src/main/java/org/elasticsearch/common/settings/Settings.java

```diff
@@ -1191,9 +1191,8 @@ public Builder put(Settings settings) {
          * @param copySecureSettings if <code>true</code> all settings including secure settings are copied.
          */
         public Builder put(Settings settings, boolean copySecureSettings) {
-            Map<String, Object> settingsMap = new HashMap<>(settings.settings);
-            processLegacyLists(settingsMap);
-            map.putAll(settingsMap);
+            map.putAll(settings.settings);
+            processLegacyLists(map);
```

Comment:
I initially just removed the `processLegacyLists(map)` call, as it felt redundant with the other call in the `build()` method. However, as I mentioned in the PR description, there is technically a change in behavior if we remove the call here, with regard to overwriting settings. I'm personally inclined to think that that's not really an issue anymore, since I don't see these legacy list settings, but I figured I'd leave the call to be sure.